pypesto/sampling/metropolis.py

```diff
@@ -29,85 +31,118 @@ def default_options(cls):
 
     def initialize(self, problem: Problem, x0: np.ndarray):
         self.problem = problem
-        self.objective = problem.objective
-        self.objective.history = History()
+        self.neglogpost = problem.objective
+        self.neglogpost.history = History()
+        if problem.x_priors is None:
+            self.neglogprior = lambda x: -0.
+        else:
+            self.neglogprior = problem.x_priors
         self.trace_x = [x0]
-        self.trace_fval = [self.objective(x0)]
+        self.trace_neglogpost = [self.neglogpost(x0)]
+        self.trace_neglogprior = [self.neglogprior(x0)]
 
-    def sample(self, n_samples: int, beta: float = 1.):
+    def sample(self, n_samples: int, beta: float = 1.,
+               temper_lpost: bool = False):
         # load last recorded particle
         x = self.trace_x[-1]
-        llh = - self.trace_fval[-1]
+        lpost = - self.trace_neglogpost[-1]
+        lprior = - self.trace_neglogprior[-1]
 
         show_progress = self.options['show_progress']
 
         # loop over iterations
         for _ in tqdm(range(int(n_samples)), disable=not show_progress):
+            kwargs = {'lpost': lpost,
+                      'lprior': lprior}
             # perform step
-            x, llh = self._perform_step(x, llh, beta)
-
+            x, lpost, lprior = self._perform_step(x, beta,
+                                                  temper_lpost,
+                                                  **kwargs)
```

Comment:
sorry, I meant
```suggestion
            x, lpost, lprior = self._perform_step(
                x=x, lpost=lpost, lprior=lprior, beta=beta,
                temper_lpost=temper_lpost)
```

Reply:
i do not get it... you suggested to use kwargs to not to mess up with the order of the inputs??

Reply:
yes, which the version I posted exactly does (it uses kwargs ...) ...